modules/videoModule/index.js

```diff
@@ -0,0 +1,209 @@
+  const videoCore = videoCore_;
+  const getConfig = getConfig_;
+  const pbGlobal = pbGlobal_;
+  const requestBids = pbGlobal.requestBids;
+  const pbEvents = pbEvents_;
+  const videoEvents = videoEvents_;
+  const gamAdServerFactory = gamAdServerFactory_;
+  let gamSubmodule;
+  const videoImpressionVerifierFactory = videoImpressionVerifierFactory_;
+  let videoImpressionVerifier;
+
+  function init() {
+    const cache = getConfig('cache');
+    videoImpressionVerifier = videoImpressionVerifierFactory(!!cache);
+    getConfig('video', ({ video }) => {
+      video.providers.forEach(provider => {
+        videoCore.registerProvider(provider);
+        videoCore.onEvents(videoEvents, (type, payload) => {
+          pbEvents.emit(videoPrefix + type, payload);
+        }, provider.divId);
+
+        const adServerConfig = provider.adServer;
+        if (!gamSubmodule && adServerConfig) {
+          gamSubmodule = gamAdServerFactory();
+        }
+      });
+    });
+
+    requestBids.before(beforeBidsRequested, 40);
+
+    pbEvents.on(CONSTANTS.EVENTS.BID_ADJUSTMENT, function (bid) {
+      videoImpressionVerifier.trackBid(bid);
+    });
+
+    pbEvents.on(videoPrefix + AD_IMPRESSION, function (payload) {
+      triggerVideoBidEvent(BID_IMPRESSION, payload);
+    });
+
+    pbEvents.on(videoPrefix + AD_ERROR, function (payload) {
+      triggerVideoBidEvent(BID_ERROR, payload);
+    });
+  }
+
+  function renderBid(divId, bid, options = {}) {
+    const adUrl = bid.vastUrl;
+    options.adXml = bid.vastXml;
+    options.winner = bid.bidder;
+    loadAdTag(adUrl, divId, options);
+  }
+
+  function getVideoOrtb(divId) {
+    return videoCore.getOrtbParams(divId);
+  }
+
+  return { init, renderBid, getVideoOrtb };
+
+  function beforeBidsRequested(nextFn, bidRequest) {
+    const adUnits = bidRequest.adUnits || pbGlobal.adUnits || [];
+    adUnits.forEach(adUnit => {
+      enrichAdUnit(adUnit);
+    });
+
+    const bidsBackHandler = bidRequest.bidsBackHandler;
+    if (!bidsBackHandler || typeof bidsBackHandler !== 'function') {
+      pbEvents.on(CONSTANTS.EVENTS.AUCTION_END, auctionEnd);
+    }
+
+    return nextFn.call(this, bidRequest);
+  }
+
+  function enrichAdUnit(adUnit) {
+    const videoMediaType = adUnit.mediaTypes.video;
+    const videoConfig = adUnit.video;
+    if (!videoMediaType || !videoConfig) {
+      return;
+    }
+
+    const oRtbParams = getVideoOrtb(videoConfig.divId);
+    if (!oRtbParams) {
+      return;
+    }
+
+    const ortbVideo = oRtbParams.video;
+    adUnit.mediaTypes.video = Object.assign({}, videoMediaType, ortbVideo);
+
+    adUnit.mediaTypes.video.context = ortbVideo.placement === 1 ? 'instream' : 'outstream';
+
+    const width = ortbVideo.w;
+    const height = ortbVideo.h;
+    if (width && height) {
+      adUnit.mediaTypes.video.playerSize = [width, height];
+    }
+
+    let ortb2 = { ortb2: mergeDeep({}, pbGlobal.getConfig('ortb2'), { site: oRtbParams.content }) };
+    pbGlobal.setConfig(ortb2);
```

Comment:
@dgirardi great observation. One of the key problems we aim to solve with the Video Module, is to automatically populate contextual information in the bid by getting the data from the source of truth, which is the video player. 

Each video provider gathers ortb data about the video player and its content. At the `requestBids` time, the information is extracted and added to `adUnit.mediaTypes.video` and `config.ortb2.site`. Note that this only happens when the ad unit's mediaType is video and the adUnit is configured to use the Video Module (indicated by populating `adUnit.video`). 
Since we only modify `adUnit.mediaTypes.video` when the ad unit references a specific video player, it seems like point 2 is addressed. 
However, point 1 is a problem. We want to support multiple players on the page, where an ad unit can be associated to a specific player. I think this makes sense since an ad unit represents a placement, and the placement is in a specific player. We want the bid request to include the contextual information in `$.site.content`. Would this work by populating `ortb2Fragments` at `startAuction` ?

Reply:
I stand corrected on point 2; I did not realize that `adUnit.video` is required.

For point 1, unfortunately I don't think there's a good way currently to manage multiple distinct `site.content`  within the same auction. I also wonder - if a separate, unrelated slot (for example a banner) is somewhere on the page, is it a problem if it gets `site.content` from one of the players on the page? 

Since v7, the guidance for adapters is to take for first party data from `bidderRequest.ortb2`, which is the same for all impressions. The same `ortb2` object is also [repeated in individual `bidRequest` objects](https://github.com/prebid/Prebid.js/blob/75818b998a683c565d31f00d47f3b96f8ad689f2/src/adapterManager.js#L232); you could hook into that logic so that the video-specific `site.content` is only added to video bid requests, and that would allow you different data for different players. 

However, that requires a refactoring of most adapters - because currently they all assume that there is only one `ortb2` shared between all impressions.

An alternative would be to propose this data somewhere else under `imp` (maybe `imp.ext.content`?); similar to [what is being discussed for `source.tid`](https://github.com/prebid/Prebid.js/issues/8573) (another field that turns out should have been defined under `imp`). 

Another alternative still could be requiring these ad units to have their own auction - hook into [ad unit validation](https://github.com/prebid/Prebid.js/blob/75818b998a683c565d31f00d47f3b96f8ad689f2/src/prebid.js#L208) to enforce that you cannot mix video players, or mix these new video slots with other media types. This way, all impressions for video auctions can use the same `site.content`, that you can set from `startAuction`. This is the easiest to implement but I'm not sure how much of a nuisance it would be for pubs. Maybe it's a good starting point?

Reply:
I think the fundamental issue is that the oRTB spec allows a bid request to have multiple impressions, but a bid request can only have 1 content. The content block is meant to describe the page (or app), not the placement per say. The description of the page is meant to include metadata from the media on the page. 

In the case where a banner imp is included, I don't think it's a problem for the bid request to include the `site.content` metadata specific to the video on the page, since it is describing the page on which the banner will be displayed. A banner doesn't have any content, but it would still be valuable for the DSPs to know what video content is on the page alongside the banner. They would potentially want to place a banner ad related to the video content.

I don't like the `imp.ext.content` because the idea is for DSPs to consume the contextual information, and since `ext`s are not standard, they would not read from there.

I think the main problem is in the case where there are multiple players on a page, and the auction includes imps for different players.

Reply:
Ortb 2.5 defines `site.content` as "Details about the Content (Section 3.2.16) within the site." - Perhaps we should add a flag on the `video` config to allow the publisher to indicate which of the video players is the main one. The use case I'm thinking of is a site with an article and a large player at the top. At the bottom of the page there might be smaller players with recommended content. 
If an auction is made with multiple imps, the main player's content will be used. Otherwise the content from the specific player associated to the ad unit will be used.